pymc3/step_methods/sgmcmc.py

```diff
@@ -0,0 +1,254 @@
+from .arraystep import Competence, ArrayStepShared
+from ..vartypes import continuous_types
+from ..model import modelcontext, inputvars
+import theano.tensor as tt
+from theano.sandbox.rng_mrg import MRG_RandomStreams as RandomStreams
+from ..theanof import join_nonshared_inputs, make_shared_replacements
+import theano
+from theano.gof.graph import inputs
+import numpy as np
+
+__all__ = ['SGFS']
+
+
+def _value_error(cond, str):
+    """Throws ValueError if cond is False"""
+    if not cond:
+        raise ValueError(str)
+
+
+def gen_random_state():
+    """Helper to generate a random state for MRG_RandomStreams"""
+    M1 = 2147483647
+    M2 = 2147462579
+    return np.random.randint(0, M1, 3).tolist() + \
+        np.random.randint(0, M2, 3).tolist()
+
+
+def elemwise_dlogp_wrt_var(model, var):
+    """Returns a function to calculate the score or gradient of the log likelihood
+    wrt to each observed datum x_i"""
+
+    terms = [
+        v.logp_elemwiset for v in model.basic_RVs if var in inputs([v.logpt])
+    ]
+    elemwise_logp = tt.add(*terms)
+
+    outputs, _ = theano.scan(\
+                lambda i, elogp, v: theano.grad(elogp[i], v),\
+                sequences=[theano.tensor.arange(elemwise_logp.shape[0])],
+                non_sequences=[elemwise_logp, var])
+
+    return outputs
+
+
+def dlogp_elemwise_and_prior(vars, shared, model):
+    """Returns a function to calculate the score or gradient of the log likelihood
+    wrt to each observed datum x_i"""
+    dlogp_elemwise = tt.as_tensor_variable(
+        [elemwise_dlogp_wrt_var(model, v) for v in vars]).T
+    dlog_prior = tt.as_tensor_variable([theano.grad(v.logpt, v) for v in vars])
+    sp_outputs, inarray = join_nonshared_inputs([dlogp_elemwise, dlog_prior],
```

Comment:
I am trying to find an example of `model.flatten_view` in the code base, can you provide one ?